File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.*;

public class Main {

	static int n;
	static int m;
	static int k;
	static int[][] graph;
	static boolean[] visited1;
	static boolean[] visited2;

	public static void main(String[] args) throws IOException {
		Main main = new Main();
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

		StringTokenizer st = null;
		st = new StringTokenizer(br.readLine());
		n = Integer.parseInt(st.nextToken());
		m = Integer.parseInt(st.nextToken());
		k = Integer.parseInt(st.nextToken());
		graph = new int[n + 1][n + 1];
		visited1 = new boolean[n + 1];
		visited2 = new boolean[n + 1];
		for (int i = 0; i < m; i++) {
			st = new StringTokenizer(br.readLine());
			int x = Integer.parseInt(st.nextToken());
			int y = Integer.parseInt(st.nextToken());
			main.add(x, y);
		}
		main.dfs(k);
		System.out.println();
		main.bfs(k);
	}

	void add(int x, int y) {
		graph[x][y] = 1;
		graph[y][x] = 1;
	}

	void dfs(int start) {
		System.out.print(start + " ");
		visited2[start] = true;
		for (int i = 1; i <= n; i++) {
			if (graph[start][i] == 1 && !visited2[i]) {
				dfs(i);
			}
		}
	}

	void bfs(int start) {
		Queue<Integer> q = new LinkedList<>();
		q.add(start);
		visited1[start] = true;
		System.out.print(start + " ");
		while (!q.isEmpty()) {
			start = q.poll();
			for (int i = 1; i < n + 1; i++) {
				if (graph[start][i] == 1 && !visited1[i]) {

					q.add(i);
					visited1[i] = true;
					System.out.print(i + " ");
				}
			}
		}

	}

}

File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.kt
import java.util.*


//연관관계 : 인접행렬로 구현

val check = BooleanArray(1001)
var n = 0
var g = Array(0){IntArray(0)}
var q: Queue<Int> = LinkedList()
fun main(args: Array<String>): Unit = with(Scanner(System.`in`)) {
     n = nextInt()
    val m = nextInt()
    val start = nextInt()
    g = Array(n+1){IntArray(n+1)}
    for(i in 0 until m){
        graph(nextInt(), nextInt()) //인접행렬로 그래프 저장
    }

    dfs(start)
    println()
    check.fill(false)
    bfs(start)

}

fun graph(a:Int, b:Int){
    g[a][b] = 1
    g[b][a] = 1
}

fun dfs(start:Int){
    check[start] = true //시작노드는 방문한 것으로
    print("$start ")
    for (i in 1..n){
        if(g[start][i] == 1 && !check[i]){
            dfs(i) // 1~n노드
        }
    }
}

fun bfs(start: Int){

    check[start] = true
    q.add(start)
    while (!q.isEmpty()){
       val start =  q.poll()
        print("$start ")

        for (i in 1..n){
            if (g[start][i] == 1 && !check[i]){
                check[i] = true
                q.add(i)
            }
        }

    }

}



